### PR TITLE
Feat keep preferences in localstorage

### PR DESCRIPTION
**Pull request by jorgenbele**
This PR introduces features that allow for persistence of values in localStorage. It does this by copying parts of contexts/states on every change to them, and then attempting to load from localstorage on initialization. 

Currently implements storage of the following:
1. Page size (for Interval)
2. Selected filter (including open/closed, acked/unacked, and If realtime/interval/never is selected).

It also stores a timestamp along side with the data, which might come in useful during eventual debugging or future improvements.

### CODE REVIEW

**Comment by jorgenbele** [comment]
Closes #208

**Review by lunkwill42 — COMMENTED** [review]
ATM I'm only getting this error when testing this branch in my browser:

```
Failed to compile

./src/components/incident/IncidentFilterToolbar.tsx
Module not found: Can't resolve '../../localstorageconsts' in '/app/src/components/incident'
```

I'm going to assume this is more npm bullshittery that I don't understand simply because I don't know the NPM ecosystem, so I will try some more...

**Comment by hmpf** [comment]
I get the same. There is no file "localstorageconsts.tsx" in `./src`. `IncidentFilterToolbar.tsx` only imports `DROPDOWN_TOOLBAR` from this file, and the string `DROPDOWN_TOOLBAR` does not exist anywhere else in the tree. I guess the file was overlooked?

Maybe `DROPDOWN_TOOLBAR` could be housed in `config.tsx` instead of a new file, but in any case, either:

1. an update of the README is needed, if we need to add a file that should not be in version control
2. if it may be in version control it needs to be added to the PR

File: src/components/incident/IncidentFilterToolbar.tsx [hunk 1]
```diff
@@ -43,6 +43,10 @@ import api, { Filter, IncidentMetadata, SourceSystem } from "../../api";
 // Config
 import { ENABLE_WEBSOCKETS_SUPPORT } from "../../config";
 
+// Utils
+import { saveToLocalStorage, fromLocalStorageOrDefault } from "../../utils";
+import { DROPDOWN_TOOLBAR } from "../../localstorageconsts";
```
Comment (hmpf) [inline src/components/incident/IncidentFilterToolbar.tsx:48]: This file does not exist.

File: src/components/filterprovider.tsx [hunk 2]
```diff
@@ -1,11 +1,15 @@
-import React, { useReducer, useContext, createContext } from "react";
+import React, { useEffect, useReducer, useContext, createContext } from "react";
 
 import { IncidentsFilter, AutoUpdate } from "../components/incidenttable/FilteredIncidentTable";
 import { Filter } from "../api";
 import { Tag, originalToTag } from "../components/tagselector";
 
 // Store
 import { ActionMap } from "../reducers/common";
+import { SELECTED_FILTER } from "../localstorageconsts";
```
Comment (hmpf) [inline src/components/filterprovider.tsx:9]: There's no no complaint here so I guess `IncidentFilterToolbar.tsx` is processed before `filterprovider,tsx`.

**Review by hmpf — DISMISSED** [review]
`localstorageconsts.tsx` is missing from the repo.

**Comment by jorgenbele** [comment]
I've added the missing file. It doesn't ever need to be changed, so it doesn't make sense to pollute the config file with it.

**Review by lunkwill42 — APPROVED** [review]
:+1:

**Pull request merged**